warpforge-visualize/src/lib.rs

```diff
@@ -0,0 +1,362 @@
+//! This code is a proof of concept for creating a graph of warpforge dependencies.
+
+// There are some useful functions and structs that are not currently used.
+// TODO: Remove the following line when cleaning up the code.
+#![allow(dead_code)]
+
+use std::collections::HashSet;
+use std::collections::VecDeque;
+use std::fmt::Display;
+use std::fmt::Formatter;
+
+use derive_more::Display;
+use indexmap::IndexMap;
+use reqwest::StatusCode;
+use reqwest::Url;
+
+use warpforge_api::plot::Plot;
+use warpforge_api::plot::PlotCapsule;
+use warpforge_api::plot::PlotInput;
+use warpforge_api::plot::Step;
+
+// Currently using my own fork of warpsys, since there are invalid json files in upstream.
+const WARPSYS_URL: &str = "https://raw.githubusercontent.com/janispeyer/warpsys/master";
```

Comment:
Okay this is the only part of this I wouldn't instantly merge ^^

We should fix the upstream data of course, but also I'd say this should probably run off the thing in `~/.warphome/catalogs/warpsys` (or perhaps any other single catalog there, if there's only one, although that's not really a behavioral convention we've standardized on yet) by default.

Reply:
Changed the URL to warptools in [7b67f10](https://github.com/warptools/rustwarp/pull/2/commits/7b67f103b7740f5c2d30891d77231f34027294df)
And fixed upstream json files: https://github.com/warptools/warpsys/pull/3

Addind other sources would also be very useful in the future for sure.